### PR TITLE
[#16] Get rid of redundant activation of parent span

### LINKED ISSUE #16: Existing parent span not deactivated when invoking CommandGateway
**Current Behavior**
If I place a command on the TracingCommandGateway with send or sendAndWait and there already is an active span before the TracingCommandGateway is invoked then it will not be deactivated. I close the span after invoking the TracingCommandGateway. If I don't invoke the TracingCommandGateway then the span is closed.

**Expected Behavior**
I expect the span that was created before invoking the TracingCommandGateway to be deactivated at the end of the try with statement. I have [attached a project](https://github.com/AxonFramework/extension-tracing/files/3317073/axon-tracing-parent-not-deactivated.tar.gz) to demonstrate the behavior. Am I doing something wrong or is this extension exhibiting undesirable behavior? If I don't invoke the CommandGateway within the try with statement then the span is deactivated.

### PR DESCRIPTION
**Pull request by jvanderkallen**
There is no need to activate the parent span. It is already active. Activating it while it is already active will cause the parent span to remain active if the scope is closed.

Fixes #16

### CODE REVIEW

File: tracing/src/main/java/org/axonframework/extensions/tracing/TracingCommandGateway.java [hunk 1]
```diff
@@ -19,12 +19,7 @@
 import io.opentracing.Span;
 import io.opentracing.Tracer;
 import io.opentracing.tag.Tags;
-import org.axonframework.commandhandling.CommandBus;
-import org.axonframework.commandhandling.CommandCallback;
-import org.axonframework.commandhandling.CommandExecutionException;
-import org.axonframework.commandhandling.CommandMessage;
-import org.axonframework.commandhandling.CommandResultMessage;
-import org.axonframework.commandhandling.GenericCommandMessage;
+import org.axonframework.commandhandling.*;
```
Comment (Sam-Kruglov) [inline tracing/src/main/java/org/axonframework/extensions/tracing/TracingCommandGateway.java]: Specific imports are better

File: tracing/src/test/java/org/axonframework/extensions/tracing/TracingCommandGatewayTest.java [hunk 2]
```diff
@@ -69,6 +67,9 @@ public void testSendWithCallback() {
         List<MockSpan> mockSpans = mockTracer.finishedSpans();
         assertThat(mockSpans.size(), is(1));
         assertThat(mockSpans.get(0).operationName(), is("sendCommandMessage"));
+
+        mockTracer.scopeManager().active().close();
```
Comment (Sam-Kruglov) [inline tracing/src/test/java/org/axonframework/extensions/tracing/TracingCommandGatewayTest.java]: Sorry, what is it you are asserting here? I am not familiar with opentracing test library, but it looks like you are calling a method on a mock object?

Also, your PR states it does not change any behavior, so it is not expected to see changes in tests.

**Comment by Sam-Kruglov** [comment]
@smcvb this has `#16` in the name but actually, it's 17 :)

**Comment by smcvb** [comment]
@Sam-Kruglov that's intentional, as this is a pull request, numbered #17, which reference issue with number 16.

**Comment by smcvb** [comment]
@jvanderkallen, me and @abuijze are slightly unsure whether we have resolved this already in a different commit. Sadly, I am unable to find such a commit too, which makes me believe we might _not_ have resolved this after all. 

Hence, I have a request for you. Would you be up for updating your Pull Request and validating whether it's still necessary?

**Comment by jvanderkallen** [comment]
@smcvb: Thanks for the update. I'll try to get on it in a timely manner.

**Comment by jvanderkallen** [comment]
@smcvb: I have merged master into the branch and tested it with Jaeger. It seems to work as expected.

**Comment by smcvb** [comment]
Thanks you very much for the verification @jvanderkallen!

**Pull request merged**